File: java/src/main/java/challenges/aoc2023/day2/service/DayTwoSolver.java
package challenges.aoc2023.day2.service;

import challenges.aoc2023.day2.model.BagGame;
import service.DaySolver;
import util.AdventUtils;

import java.util.List;

public class DayTwoSolver implements DaySolver {
    private final List<String> inputLines = AdventUtils.parseInput(2023, 2);
    private final BagGameService bagGameService;

    public DayTwoSolver(BagGameService bagGameService) {
        this.bagGameService = bagGameService;
    }

    public String taskOne() {
        final List<BagGame> bagGames = inputLines.stream()
            .map(BagGame::of)
            .toList();

        return bagGameService.calculateTotalOfPossibleIds(bagGames);
    }

    public String taskTwo() {
        return null;
    }
}

File: scala/src/main/scala/challenges/aoc2025/day2/service/DayTwoSolver.scala
package challenges.aoc2025.day2.service

import challenges.aoc2025.day2.model.IdRange
import service.DaySolver
import util.AdventUtils

class DayTwoSolver(validationService: IdValidationService) extends DaySolver {
  private val inputLine: String = AdventUtils.parseInput(2025, 2).head

  override def taskOne(): String = {
    val ranges: List[IdRange] = inputLine.split(",")
      .map(IdRange.of)
      .toList
    
    validationService.calculateInvalidIdTotal(ranges).toString
  }

  override def taskTwo(): String = ???
}
